misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/mock/GenericMockObjectValue.java

```diff
@@ -226,7 +227,7 @@ private String getCaseIds(BObject mockObj, String fieldName) {
         List<Object> newArgs = new ArrayList<>();
         int i = 0;
         while (i < args.length) {
-            if (args[i] != null) {
+            if (args[i] != null && !(args[i] instanceof TypedescValue)) {
```

Comment:
Shall we check if there is a proper API to get the typedesc during the runtime? @MaryamZi

Reply:
I think it would be better if we pass `args[i]` to `io.ballerina.runtime.api.utils.TypeUtils#getType` and then check if the type tag of the returned type is not equal to `io.ballerina.runtime.api.TypeTags#TYPEDESC_TAG`.

Reply:
I implemented the check as such :`TypeUtils.getType(args[i]).getTag() != TypeTags.TYPEDESC_TAG`